libpimeval/src/pimSim.cpp

```diff
@@ -82,7 +123,9 @@ bool
 pimSim::createDevice(PimDeviceEnum deviceType, unsigned numRanks, unsigned numBankPerRank, unsigned numSubarrayPerBank, unsigned numRows, unsigned numCols)
 {
   pimPerfMon perfMon("createDevice");
+  init();
```

Comment:
We need to move down this init call after the check below. If there is already a device, we should just return false without calling init or uninit.

Reply:
Thanks. Fixed!